File: fastapi/app/schemas/staff.py
# app/schemas/staff.py
from pydantic import BaseModel
from typing import List

class StaffAttributeBase(BaseModel):
    name: str
    name_ruby: str
    mail_address: str

class StaffAttributeResponseSchema(BaseModel):
    name: str
    name_ruby: str
    mail_address: str

    class Config:
        orm_mode = True

class StaffCreateSchema(BaseModel):
    role_id: int
    staff_attributes: StaffAttributeBase
    password: str

class StaffUpdateSchema(BaseModel):
    name: str
    role: str

class StaffResponseSchema(BaseModel):
    id: int
    role_id: int
    store_id: int
    staff_attributes: List[StaffAttributeResponseSchema]

    class Config:
        orm_mode = True

File: nextjs/src/type/staff.ts
// src/type/staff.ts
export interface StaffBase {
  role_id: number;
  store_id: number;
}

export interface StaffCreate extends StaffBase {
  password: string;
  staff_attributes: StaffAttributeBase;
}

export interface Staff extends StaffBase {
  id: number;
  staff_attributes: StaffAttribute[];
}

export interface StaffAttributeBase {
  name: string;
  name_ruby: string;
  mail_address: string;
}

export interface StaffAttribute extends StaffAttributeBase {
  id: number;
  staff_id: number;
}